### ISSUE TITLE
[NV-2380] bug: user can access prod environment using dev api key

### ISSUE THREAD

**Issue by jainpawan21** [novuhq/novu#3490]
### Discussed in https://github.com/orgs/novuhq/discussions/3483

<div type='discussions-op-text'>

<sup>Originally posted by **leonardbinet** May 23, 2023</sup>
Hi there 👋,

Given 2 environments, `dev` and `production` (default), each environment has its dedicated api key.

I've noticed that by calling: (https://docs.novu.co/api/get-environments/)
```
curl -X GET \
 -H "Authorization: ApiKey <dev api key>" \
https://api.novu.co/v1/environments
```

I would get all environments info, with related api keys (both `production` and `dev`) 🙀 

Long story short, with `dev` api key I get access to `production` environment.

It seems to me like a serious security flaw, why would you want api keys to be returned in this case?</div>

<sub>[NV-2380](https://linear.app/novu/issue/NV-2380/bug-user-can-access-prod-environment-using-dev-api-key)</sub>

**Comment by michaldziuba03** [comment]
@jainpawan21 can you assign me to this issue?

**Comment by jainpawan21** [comment]
Sure

**Comment by gitstart** [comment]
@jainpawan21 @scopsy I would like to pick this up

**Comment by jainpawan21** [comment]
@michaldziuba03 how is it going with this issue?

**Comment by michaldziuba03** [comment]
@jainpawan21 I work on this issue
